### DISCUSSION TITLE
External MIDI button/slider options in OpenControl Editor

### DISCUSSION THREAD

**Discussion by ll-dasein-ll** [KBLiveSolutions/open.control#27] [Feature Requests & Ideas]
Hi -

I was wondering how easy it would be to customise the opencontrol editor page to change the number of available buttons and sliders in the bottom External MIDI section?  I've got a couple of MIDI controllers and would like to be able to add and/or remove the number of buttons and sliders that can be configured in the editor depending on which one I'm using.  Would this be possible, or is there more stuff going on in the background than it just being a simple task of adding more fields to the External MIDI section?

Thanks in advance!

**Comment by KBLiveSolutions** [comment]
Hi

It's a bit complicated, it's not just the number of boxes that appear in the editor, it's also the number of "slots" available inside open·control.
So that would require to change the firmware.

Can you give me a concrete example of what you're trying to do ? Maybe there s another solution.

**Reply by ll-dasein-ll** [reply]
Hi Pierre - thanks for getting back to me, but please ignore this question!  I had a look through the user manual to remind myself of something else last night, and have realised that I'd completely misunderstood how the external MIDI functionality worked, so it's fine :-).

I think the fact that there are 8 buttons listed threw me, because I wasn't thinking of the two encoders as buttons too, so I thought external devices could be used to act as additional controls on top of the 6 main opencontrol buttons and dials, and wondered if more could be added.  I hope that makes sense!  I feel a bit stupid for asking now that I've realised my mistake!

Thanks for your time though!
